Zend/zend_execute.c

```diff
@@ -775,16 +775,17 @@ static bool zend_verify_weak_scalar_type_hint_no_sideeffect(uint32_t type_mask,
 	double dval;
 	bool bval;
 
-	if ((type_mask & MAY_BE_LONG) && zend_parse_arg_long_weak(arg, &lval, 0)) {
+	/* Pass (uint32_t)-1 as arg_num to indicate to ZPP not to emit any deprecation notice */
```

Comment:
Not that it really matters, but as argument numbers are 1-based, you could have kept the 0 value ^^